### DISCUSSION TITLE
Hazard detection node

### DISCUSSION THREAD

**Discussion by gmptrr98** [iRobotEducation/create3_docs#203] [Questions] [answered]
Hello! I'm working on a node subscribed to HazardDetectionVector. What I would like to do is simply to detect a bump and move accordingly but for some reason I can't seem to unpack the information needed from the message. I would just need to get the correct frame id and the error type, below is my code. 

   
```
def listener_callback_hazard_detection(self, hazard_detection):

        twist=Twist()
        if str(hazard_detection.detections.type)== 1:
            self.get_logger().info('Ouch')            
            if str(hazard_detection.header.frame_id) == ('bump_front_left' or 'bump_left'):
                twist.angular.z=1.0
                self.get_logger().info('Ouch')   
                self.publisher1.publish(twist)   
            elif str(hazard_detection.header.frame_id) == ('bump_front_right' or 'bump_right'):
                self.get_logger().info('Yuck')   
                twist.angular.z=1.0
                self.publisher1.publish(twist)   
```

**Comment by justinIRBT** [comment, marked as answer]
detections is a vector inside the hazard detection message.  The frame of the hazard is inside the header of the object in the vector.  For something along these lines, see the vector usage in the callback method (this is untested code, so may not work as is):
```
from irobot_create_msgs.msg import HazardDetectionVector
class MinimalSubscriber(Node):

    def __init__(self):
        super().__init__('minimal_subscriber')
        self.subscription = self.create_subscription(
            HazardDetectionVector,
            'hazard_detection',
            self.listener_callback,
            rclpy.qos.qos_profile_sensor_data)
        self.subscription  # prevent unused variable warning

    def listener_callback(self, msg):
        for detection in msg.detections:
            if detection.type == 1:
                self.get_logger().info('Ouch')
                if detection.header.frame_id == 'bump_front_left':
                    self.get_logger().info('Ouch Left')
```

**Reply by gmptrr98** [reply]
Ok, it's almost working, now I have the following problem, I'm using this code:

```
    def listener_callback_hazard_detection(self, hazard_detection):

        for detection in hazard_detection.detections:
            if detection.type== 1:          
                if  detection.header.frame_id == 'bump_front_left' or 'bump_left' :
                    self.get_logger().info(detection.header.frame_id)   
                    self.get_logger().info('Yuck, bumped on the left')      
                elif detection.header.frame_id == 'bump_front_right' or 'bump_right':
                    self.get_logger().info('Ouch, bumped on the right')   
  
```

For some reason I'm getting the following output in the terminal:

![Screenshot from 2022-09-19 13-58-56](https://user-images.githubusercontent.com/110682193/191083670-015045db-cac5-491e-baa8-99d212947024.png)

So regardless of the frame_id I always enter in the first if condition. Why is that? How can I fix this? Thank you!

**Reply by justinIRBT** [reply]
Your == comparison is only being compared to `detection.header.frame_id == 'bump_front_left'`, the or condition is being applied separately, which always evaluates to `true`.

You need `detection.header.frame_id == 'bump_front_left' or detection.header.frame_id == 'bump_left'`

**Reply by gmptrr98** [reply]
I see, it's working now! 
Thank you very much :)
